gcloud-java-dns/src/test/java/com/google/gcloud/dns/testing/LocalDnsHelperTest.java

```diff
@@ -0,0 +1,1429 @@
+  private static final ResourceRecordSet RRSET2 = new ResourceRecordSet();
+  private static final ResourceRecordSet RRSET_KEEP = new ResourceRecordSet();
+  private static final String PROJECT_ID1 = "2135436541254";
+  private static final String PROJECT_ID2 = "882248761325";
+  private static final String ZONE_NAME1 = "my little zone";
+  private static final String ZONE_NAME2 = "another zone name";
+  private static final ManagedZone ZONE1 = new ManagedZone();
+  private static final ManagedZone ZONE2 = new ManagedZone();
+  private static final String DNS_NAME = "www.example.com.";
+  private static final Change CHANGE1 = new Change();
+  private static final Change CHANGE2 = new Change();
+  private static final Change CHANGE_KEEP = new Change();
+  private static final Change CHANGE_COMPLEX = new Change();
+  private static final LocalDnsHelper LOCAL_DNS_HELPER = LocalDnsHelper.create(0L); // synchronous
+  private static final Map<DnsRpc.Option, ?> EMPTY_RPC_OPTIONS = ImmutableMap.of();
+  private static final DnsRpc RPC =
+      new DefaultDnsRpc(LOCAL_DNS_HELPER.options());
+  private static final String REAL_PROJECT_ID = LOCAL_DNS_HELPER.options().projectId();
+  private Map<String, Object> optionsMap;
+
+  @BeforeClass
+  public static void before() {
+    ZONE1.setName(ZONE_NAME1);
+    ZONE1.setDescription("");
+    ZONE1.setDnsName(DNS_NAME);
+    ZONE1.setNameServerSet("somenameserverset");
+    ZONE2.setName(ZONE_NAME2);
+    ZONE2.setDescription("");
+    ZONE2.setDnsName(DNS_NAME);
+    ZONE2.setNameServerSet("somenameserverset");
+    RRSET1.setName(DNS_NAME);
+    RRSET1.setType(RRSET_TYPE);
+    RRSET1.setRrdatas(ImmutableList.of("1.1.1.1"));
+    RRSET2.setName(DNS_NAME);
+    RRSET2.setType(RRSET_TYPE);
+    RRSET2.setRrdatas(ImmutableList.of("123.132.153.156"));
+    RRSET_KEEP.setName(DNS_NAME);
+    RRSET_KEEP.setType("MX");
+    RRSET_KEEP.setRrdatas(ImmutableList.of("255.255.255.254"));
+    CHANGE1.setAdditions(ImmutableList.of(RRSET1, RRSET2));
+    CHANGE2.setDeletions(ImmutableList.of(RRSET2));
+    CHANGE_KEEP.setAdditions(ImmutableList.of(RRSET_KEEP));
+    CHANGE_COMPLEX.setAdditions(ImmutableList.of(RRSET_KEEP));
+    CHANGE_COMPLEX.setDeletions(ImmutableList.of(RRSET_KEEP));
+    LOCAL_DNS_HELPER.start();
+  }
+
+  @Before
+  public void setUp() {
+    resetProjects();
+    optionsMap = new HashMap<>();
+  }
+
+  private static void resetProjects() {
+    for (String project : LOCAL_DNS_HELPER.projects().keySet()) {
+      LOCAL_DNS_HELPER.projects().remove(project);
+    }
+  }
+
+  @AfterClass
+  public static void after() {
+    LOCAL_DNS_HELPER.stop();
+  }
+
+  @Test
+  public void testCreateZone() {
+    ManagedZone created = RPC.create(ZONE1, EMPTY_RPC_OPTIONS);
+    // check that default records were created
+    DnsRpc.ListResult<ResourceRecordSet> listResult
+        = RPC.listDnsRecords(ZONE1.getName(), EMPTY_RPC_OPTIONS);
+    assertEquals(2, Lists.newLinkedList(listResult.results()).size());
```

Comment:
It would be nice to assert that the two records are equal the the SOA and NS records rather than just checking the size.